tapdb/assets_store.go

```diff
@@ -3088,9 +3088,11 @@ func (a *AssetStore) LogAnchorTxConfirm(ctx context.Context,
 		// We'll keep around the IDs of the assets that we set to being
 		// spent. We'll need one of them as our template to create the
 		// new assets.
-		spentAssetIDs := make([]int64, len(inputs))
+		spentAssetIDs := make(map[asset.ID]int64, len(inputs))
```

Comment:
Isn't this also vulnerable to a collision? If we spend multiple inputs that are the same asset ID, but distinct script key.

Reply:
Yes, there can be collisions if we spend multiple inputs of the same asset ID.
But for this case it doesn't matter, since we just need one template to create a copy from. And as long as that template comes from the correct asset ID (making sure that the references to the genesis asset and group key are correct to be copied), it doesn't matter _which_ one we use. Clarified this with comments and by renaming the variable to `copyTemplateIDs`.